toothpick-runtime/src/main/java/toothpick/ToothPick.java

```diff
@@ -52,14 +57,17 @@ public static Scope openScopes(Object... names) {
    * Otherwise a new scope is created.
    */
   public static Scope openScope(Object name) {
-    synchronized (ToothPick.class) {
-      Scope scope = MAP_KEY_TO_SCOPE.get(name);
-      if (scope == null) {
-        scope = new ScopeImpl(name);
-        MAP_KEY_TO_SCOPE.put(name, scope);
-      }
+    Scope scope = MAP_KEY_TO_SCOPE.get(name);
+    if (scope != null) {
```

Comment:
We have:
... A
../ ...\
B.....C

And then:
1. Thread 1 calls -> closeScope(A)
2. Thread 2 calls -> openScope(C)

Thread 2 can get scope C when it is about to be removed. Should we allow that?